shub_image/deploy.py

```diff
@@ -125,9 +127,12 @@ def _prepare_deploy_params(project, version, image_name, endpoint, apikey,
     if not username:
         params['pull_insecure_registry'] = True
     else:
+        # Missing password leads to auth request to registry authentication
+        # service without 'account' query parameter which breaks login
+        # procedure.
         params['pull_auth_config'] = json.dumps(
             {'username': username,
-             'password': password,
+             'password': password or ' ',
```

Comment:
I don't think we can substitute is like that. It works for internal registry login, because it requires apikey as username, but here user can specify external registry, like docker hub, isn't it?